src/file-upload/file-list.tsx

```diff
@@ -0,0 +1,237 @@
+import findIndex from "lodash/findIndex";
+import isEqual from "lodash/isEqual";
+import { useEffect, useState } from "react";
+import { useResizeDetector } from "react-resize-detector";
+import { FileItem } from "./file-item";
+import { FileItemEdit } from "./file-item-edit";
+import { EditableItemsContainer, ListWrapper } from "./file-list.styles";
+import { FileUploadHelper } from "./helper";
+import { FileItemProps } from "./types";
+import { SimpleIdGenerator } from "../util";
+
+// =============================================================================
+// INTERFACES
+// =============================================================================
+type RenderMode = "cancelled-edit" | "list" | "edit";
+
+interface FileItemRenderProps {
+    fileItem: FileItemProps;
+    renderMode: RenderMode;
+}
+
+type RenderItem = FileItemRenderProps | FileItemRenderProps[];
+
+interface Props {
+    fileItems: FileItemProps[];
+    onItemUpdate: (item: FileItemProps) => void;
+    onItemDelete: (item: FileItemProps) => void;
+    editableFileItems: boolean;
+    descriptionMaxLength?: number | undefined;
+}
+
+// =============================================================================
+// COMPONENT
+// =============================================================================
+
+export const FileList = ({
+    fileItems,
+    editableFileItems,
+    descriptionMaxLength,
+    onItemUpdate,
+    onItemDelete,
+}: Props) => {
+    // =========================================================================
+    // CONST, STATE, REFS
+    // =========================================================================
+    const [arrangedItems, setArrangedItems] = useState<RenderItem[]>([]);
+    const [itemsRenderMode, setItemsRenderMode] = useState<
+        FileItemRenderProps[]
+    >([]);
+
+    const { width: wrapperWidth, ref: wrapperRef } = useResizeDetector();
+
+    // =========================================================================
+    // EFFECTS
+    // =========================================================================
+    useEffect(() => {
+        setItemsRenderMode(getItemsRenderMode(fileItems));
+    }, [fileItems]);
+
+    useEffect(() => {
+        setArrangedItems(getArrangedItemsToRender(itemsRenderMode));
+    }, [itemsRenderMode]);
+
+    // =========================================================================
+    // EVENT HANDLERS
+    // =========================================================================
+    const handleDescriptionUpdate =
+        (item: FileItemProps) => (description: string) => {
+            const updatedItem = { ...item, description };
+            onItemUpdate(updatedItem);
```

Comment:
can the item render mode be immediately updated to `list` here if description is not undefined?